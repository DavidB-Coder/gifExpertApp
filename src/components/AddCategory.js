import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim()) {
            setCategory( ps => [inputValue,...ps] );        
            setInputValue('');
        }
    }


    return (
        <>
            <form onSubmit={ handleSubmit }>
                <input
                    type="text"
                    value={ inputValue }                
                    onChange={(e) => {setInputValue(e.target.value)}}
                    placeholder={'Ingresa una nueva categorÃ­a'}
                />
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategory : PropTypes.func.isRequired
};
