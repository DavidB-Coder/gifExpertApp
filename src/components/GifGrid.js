import React from 'react'
import { useFetchGifs } from '../Hooks/useFetchGifs'
import { GifGridItem } from './GifGridItem'
import PropTypes from 'prop-types'

export const GifGrid = ({category}) => {

    const {data, loading} = useFetchGifs( category );
    
    return (
        <>
            { loading && <p className="animate__animated animate__flash">Loading...</p> }
            <h3>{category}</h3>                        
            <div className="card-grid">
                {data.map((img) => {
                    return(
                        <GifGridItem 
                            key={img.id}                         
                            {...img}
                        />
                    )
                })}
            </div>            
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}