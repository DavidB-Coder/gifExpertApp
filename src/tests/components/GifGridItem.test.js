import React from 'react'
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('pruebas en GifGridItem', () => {

    const title = "un título";
    const url = "https://url.com.co";
    const wrapper = shallow(<GifGridItem title={title} url={url}/>) 

    test('debe de mostrar el componente correctamente', () => {
        expect( wrapper ).toMatchSnapshot()       
    })

    test('debe tener un párrafo con el título', () => {
        const p = wrapper.find('p');
        expect( p.text().trim() ).toBe( title );
    })
    
    test('debe tener la imagen igual al url y alt de los props', () => {
        
        const img = wrapper.find( 'img' );
        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)

    })
    
    test('debe tener clase animate__fadeIn', () => {
        
        const div = wrapper.find('div');
        expect(div.hasClass('animate__bounceIn')).toBe(true)

    })
    
    
    

})
