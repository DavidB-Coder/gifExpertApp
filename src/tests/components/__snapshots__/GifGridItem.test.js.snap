// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="un título"
    src="https://url.com.co"
  />
  <p>
    un título
  </p>
</div>
`;
